compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorize.cpp

```diff
@@ -50,8 +50,10 @@ namespace iree_compiler {
 namespace {
 
 int getComputeVectorSize(int64_t size) {
-  // Try to use 4 first, and then 2, and then 1.
-  return size % 4 == 0 ? 4 : (size % 2 == 0 ? 2 : 1);
+  for (int i : {4, 3, 2}) {
+    if (size % i == 0) return i;
```

Comment:
I guess we want to support because we have a lowering path through `vector.reduce vector ... : <3x...>` -> `spirv.SDotAccSat`? Does it also work for the other flow, i.e., outer product matmul lowering?

Reply:
I'll separate out this change to a new pull request later.

Reply:
SG!

Reply:
Created https://github.com/openxla/iree/pull/13642 for this.